src/editor/serialize.ts

```diff
@@ -31,7 +32,7 @@ export function mdSerialize(model: EditorModel) {
                 return html + part.text;
             case "room-pill":
             case "user-pill":
-                return html + `[${part.text.replace(/[[\\\]]/g, c => "\\" + c)}](${makeGenericPermalink(part.resourceId)})`;
+                return html + `[${part.text.replace(/[[\\\]]/g, c => "\\" + c)}](${makeGenericPermalink((part as PillPart).resourceId)})`;
```

Comment:
Is this the only place we expect to switch over part types? Is it worth expressing them as a discriminated union?

Reply:
That is a grand idea!

Reply:
okay that was a PAIN because class extensions don't play well with it :D

Reply:
Sorry for putting you through it

Reply:
worth it in the long term